src/OnboardingSPA/pages/Steps/GetStarted/SiteTypeSetup/PrimarySite/index.js

```diff
@@ -37,130 +37,138 @@ const StepPrimarySetup = () => {
 	};
 
+	const primarySiteTypeChips = () => {
+		let idx = 0;
+		const primaryChipList = [];
+
+		const types = siteClassData?.types;
+		for ( const typeKey in types ) {
+			primaryChipList.push(
+				<div
+					key={ types[ typeKey ]?.slug }
+					tabIndex={ idx + 1 }
+					role="button"
+					className={ `${
+						types[ typeKey ].slug === primaryCategory
+							? 'chosenPrimaryCategory '
+							: ''
+					}nfd-card-pri-category` }
+					onClick={ () =>
+						handleCategoryClick( types[ typeKey ].slug )
+					}
+					onKeyDown={ () =>
+						handleCategoryClick( types[ typeKey ].slug )
+					}
+				>
+					<div className="nfd-card-pri-category-wrapper">
+						<span
+							className={ `nfd-card-pri-category-wrapper-icon ${
+								types[ typeKey ].slug === primaryCategory
+									? 'nfd-card-pri-category-wrapper-icon-selected '
+									: ''
+							}` }
+							style={ {
+								backgroundImage: `url(${ types[ typeKey ]?.icon })`,
+							} }
+						></span>
+						<span className="categName">
+							{ types[ typeKey ]?.label }
+						</span>
+					</div>
+				</div>
+			);
+			idx++;
+		}
+
+		return primaryChipList;
+	};
+
 	return (
 		<CommonLayout isBgPrimary isCentered>
 			<NewfoldLargeCard>
 				<div className="nfd-card-heading center">
 					<CardHeader
-						heading={ __(
-							currentStep?.heading,
-							'wp-module-onboarding'
-						) }
-						subHeading={ sprintf(
-							__( content.subHeading, 'wp-module-onboarding' ),
-							translations( 'SITE' )
-						) }
-						question={ __(
-							currentStep?.subheading,
-							'wp-module-onboarding'
-						) }
+						heading={ contents.cardHeading }
+						subHeading={ contents.subHeading }
+						question={ contents.question }
 					/>
 				</div>
-				<Animate
-					type="fade-in-disabled"
-					after={
-						content.categories &&
-						selectedPrimaryCategoryInStore !== null
-					}
-				>
+				<Animate type="fade-in-disabled" after={ siteClassData }>
 					<div className="nfd-setup-primary-categories">
-						{ content.categories.map( ( item, idx ) => {
-							return (
-								<div
-									key={ item?.name }
-									className={ `${
-										clickedIndex === idx ||
-										item.name ===
-											selectedPrimaryCategoryInStore
-											? 'chosenPrimaryCategory '
-											: ''
-									}nfd-card-category` }
-									onClick={ ( e ) =>
-										handleCategoryClick( idx )
-									}
-								>
-									<div className="nfd-card-category-wrapper">
-										<span
-											className="icon"
-											style={ {
-												backgroundImage:
-													clickedIndex !== idx &&
-													item.name !==
-														selectedPrimaryCategoryInStore
-														? item?.icon
-														: item?.iconWhite,
-											} }
-										></span>
-										<span className="categName">
-											{ item?.name }
-										</span>
-									</div>
-								</div>
-							);
-						} ) }
+						{ siteClassData && primarySiteTypeChips() }
```

Comment:
Do we need to check `siteClassData` here? Doesn't Animate render children only after `siteClassData` exists?

Reply:
No, It doesn't work as the function runs on undefined at first, and later animate just shows it later when the data comes and doesn't stop the code from running.

Reply:
Lazy load seems to be not for non-functional components

Reply:
Cool, small enough case so let's figure this out some other time when a blocking use case actually arises, looks good for now.